packages/qwik/src/optimizer/src/plugins/rollup.ts

```diff
@@ -201,9 +216,32 @@ export function normalizeRollupOutputOptionsObject(
         ? `${opts.assetsDir}/${assetFileNames}`
         : assetFileNames;
     }
-    // Friendly name in dev or preview with debug mode
-    const fileName =
-      opts.buildMode == 'production' && !opts.debug ? 'build/q-[hash].js' : 'build/[name].js';
+
+    let fileName: string | ((chunkInfo: Rollup.PreRenderedChunk) => string) | undefined;
+    if (opts.buildMode === 'production' && !opts.debug) {
+      fileName = 'build/q-[hash].js';
+    } else {
+      // Friendlier names in dev or preview with debug mode
+      fileName = (chunkInfo) => {
+        if (chunkInfo.moduleIds.some((id) => id.endsWith('core.prod.mjs'))) {
+          return 'build/core.js';
+        }
+        if (chunkInfo.moduleIds.some((id) => id.endsWith('qwik-city/lib/index.qwik.mjs'))) {
+          return 'build/qwik-city.js';
+        }
+
+        // some chunk names are absolute paths and rollup doesn't accept that so we must sanitize those instead of simply returning `q-[name]`.
+
+        const path = optimizer.sys.path;
```

Comment:
question: is it necessary to use the `optimizer.sys.path`? I used it here for consistency.

Reply:
seems prudent